File: Java/Banking-java-dnbc5/src/banking/Account.java
package banking;

import banking.exceptions.DepositWithdrawTransferAmountNotPositiveException;
import banking.exceptions.InsufficientFundsException;

public class Account {

	private static int nextId = 1;
	
	private int id;
	private double balance;
	private String description;
	
	// in C#: public int id { get; set; }
	public int getId() { return id; }
	// io C#: public double Balance { get; private set; }
	public double getBalance() { return balance; }
	private void setBalance(double newBalance) { balance = newBalance; }
	
	public String getDescription() { 
		return description; 
	}
	public void setDescription(String newDescription) { 
		description = newDescription; 
	}
	public boolean transferTo(Account account, double amount) throws DepositWithdrawTransferAmountNotPositiveException {
		try {
			withdraw(amount);
		} catch(InsufficientFundsException ex) {
			return false;
		}
		account.deposit(amount);
		return true;
	}
	private void checkAmountGtZero(double amount) throws DepositWithdrawTransferAmountNotPositiveException {
		if(amount <= 0) {
			throw new DepositWithdrawTransferAmountNotPositiveException("Amount must be positive.");
		}
	}
	public void deposit(double amount) throws DepositWithdrawTransferAmountNotPositiveException {
		checkAmountGtZero(amount);
		setBalance(getBalance() + amount);
	}
	public void withdraw(double amount) 
			throws DepositWithdrawTransferAmountNotPositiveException, InsufficientFundsException {
		checkAmountGtZero(amount);
		if(getBalance() < amount) {
			throw new InsufficientFundsException("Insufficient funds.");
		}
		setBalance(getBalance() - amount);
	}
	
	// in C#: public Account(string NewDescription) : this()
	public Account(String newDescription) {
		this();
		description = newDescription;
	}

	public Account() {
		id = nextId++;
		balance = 0;
		description = "*New Account*";
	}
}

File: BankingSolution/BankingProject/Account.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BankingProject {
    class Account : IComparable<Account>, IAccount {

        private static int nextId = 1;

        private int Id { get; set; }
        private string Description { get; set; }
        private decimal Balance { get; set; }

        public int CompareTo(Account acct) {
            if(this.Description.Equals(acct.Description)) {
                return 0;
            }
            if(this.Description.CompareTo(acct.Description) > 0) {
                return 1;
            } else {
                return -1;
            }
        }

        public void TransferTo(decimal Amount, Account Acct) {
            var BalanceBeforeWithdraw = GetBalance();
            Withdraw(Amount);
            var BalanceAfterWithdraw = GetBalance();
            if(BalanceBeforeWithdraw == BalanceAfterWithdraw) {
                Console.WriteLine("Insufficient Funds");
                return;
            }
            Acct.Deposit(Amount);
        }

        public int GetId() {
            return Id;
        }
        public string GetDescription() {
            return Description;
        }
        public void SetDescription(string NewDescription) {
            Description = NewDescription;
        }
        public decimal GetBalance() {
            return Balance;
        }
        public void Deposit(decimal Amount) {
            if(Amount <= 0) {
                Console.WriteLine("Amount must be positive.");
            } else {
                Balance += Amount;
            }
        }
        public void Withdraw(decimal Amount) {
            if(Amount <= 0) {
                Console.WriteLine("Amount must be positive.");
                return;
            }
            if(Amount > Balance) {
                //Console.WriteLine("Insufficient Funds!");
                //return;
                throw new InsufficientFundsException();
            }
            Balance -= Amount;
        }
        public virtual string Print() {
            return $"Id={Id}, Description={Description}, Balance={Balance}";
        }
        public Account(string NewDescription, int i) {
            Id = nextId++;
            if(NewDescription == null) {
                Description = "*New Account*";
            } else {
                Description = NewDescription;
            }
            Balance = 0;
        }
        // This constructor calls the other contructor
        // and passes in a null value for NewDescription
        public Account() : this(null, 0) {
        }
    }
}
